src/librustc_mir/build/scope.rs

```diff
@@ -828,6 +879,27 @@ fn build_diverge_scope<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>,
         };
     }
 
+    let repatching;
+    if scope.diverge_path.terminator_emitted {
+        // we need to re-patch the terminator so that it now points to
+        // any drops that were inserted above.
+        cfg.block_data_mut(scope.diverge_path.block).terminator = None;
+        scope.diverge_path.repatch_count += 1;
+        repatching = format!("repatch {}", scope.diverge_path.repatch_count);
```

Comment:
maybe make this a closure to avoid creating a string when not emitting `debug!`?